remote/mdb_service_scripting.go

```diff
@@ -18,11 +23,198 @@ const (
 	ScriptingTestCommand      = "test_scripting"
 )
 
-func (s *mdbService) scriptingGet(ctx context.Context, w io.Writer, msg mongowire.Message)       {}
-func (s *mdbService) scriptingCreate(ctx context.Context, w io.Writer, msg mongowire.Message)    {}
-func (s *mdbService) scriptingSetup(ctx context.Context, w io.Writer, msg mongowire.Message)     {}
-func (s *mdbService) scriptingCleanup(ctx context.Context, w io.Writer, msg mongowire.Message)   {}
-func (s *mdbService) scriptingRun(ctx context.Context, w io.Writer, msg mongowire.Message)       {}
-func (s *mdbService) scriptingRunScript(ctx context.Context, w io.Writer, msg mongowire.Message) {}
-func (s *mdbService) scriptingBuild(ctx context.Context, w io.Writer, msg mongowire.Message)     {}
-func (s *mdbService) scriptingTest(ctx context.Context, w io.Writer, msg mongowire.Message)      {}
+func (s *mdbService) scriptingGet(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &scriptingGetRequest{}
+	if !s.serviceScriptingRequest(ctx, w, msg, req, ScriptingGetCommand) {
+		return
+	}
+
+	harness := s.getHarness(ctx, w, req.ID, ScriptingGetCommand)
+	if harness == nil {
+		return
+	}
+
+	s.serviceScriptingResponse(ctx, w, nil, ScriptingGetCommand)
+}
+
+func (s *mdbService) scriptingCreate(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	req := &scriptingCreateRequest{}
+	if !s.serviceScriptingRequest(ctx, w, msg, req, ScriptingCreateCommand) {
+		return
+	}
+
+	opts, err := options.NewScriptingHarness(req.Params.Type)
+	if err != nil {
+		shell.WriteErrorResponse(ctx, w, mongowire.OP_REPLY, errors.Wrap(err, "problem creating harness options"), ScriptingCreateCommand)
+	}
+	if err = bson.Unmarshal(req.Params.Options, opts); err != nil {
```

Comment:
See comment about scriptingCreateRequest.

Reply:
does the rest of the mrdb service use the bson library?

Reply:
Yes. It's because birch doesn't have an equivalent `bson.Marshal` function and there's no easy way to make a birch document from an arbitrary struct without first serializing it to BSON in the first place.